app/services/generate_vendor_providers.rb

```diff
@@ -13,10 +13,28 @@ def self.call
       { name: 'Gordon Associates Provider', code: 'GORD' },
       { name: 'University of Newcastle Upon Tyne', code: 'N21' },
       { name: 'University College Birmingham', code: 'B35' },
+      { name: 'Test Provider', code: 'TEST' },
     ]
 
     providers.each do |provider|
       Provider.find_or_create_by(provider)
     end
+
+    providers.reject { |p| p[:code] == 'TEST' }.each do |provider|
```

Comment:
Could we assign `Provider.find_by(code: 'TEST')` to a variable and pass it to `generate_courses_as_an_accredited_body` as `accredited_body_provider` (or whatever name makes sense).
This will mean we don't need to `find_by` within the method.

```ruby
accredited_body_provider = providers.find { |p| p[:code] == 'TEST' }

providers.except(accredited_body_provider).each do |provider|
  generate_courses_as_an_accredited_body(provider, accredited_body_provider)
end
```